Signal/src/util/Searcher.swift

```diff
@@ -3,6 +3,91 @@
 //
 
 import Foundation
+import SignalServiceKit
+
+@objc
+class ConversationSearcher: NSObject {
```

Comment:
Should each Searcher have its own source file?

Reply:
Sure, I'll move ConversationSearcher into it's own file. I'm going to leave AnySearcher with Searcher since it's just a thing wrapper for objc compatibility.